### ISSUE TITLE
Bug: @typescript-eslint/naming-convention doesn't fail all uppercase enum member when PascalCase

### ISSUE THREAD

**Issue by xenoterracide** [typescript-eslint/typescript-eslint#5486]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=FAUwHgDg9gTgLgAhAOwK4FsEGVZwCICWMIAxnAVMggN7AIICCAziQgLwIDkDWAwpwBo6CPCBbsueAKJ9BwAL5A&eslintrc=N4KABGBECmAeAu0B2ATAzpAXGA2uCU0aANgJZLyYBO0AxgPYC2jyK0Kk+AugDT6QAHYgFcA5uQzY8BKAAF4ATwFFaVUgPgBaImQqcIvflWHEiWMKBmR5SlWo3aS5eAHokAQ0blRmhkgBuyPCk9EjmOIRUVPRUkHwywGD4VmjQprTwMeYwSMKMALLQjABG0LHxMlAAZjGM7vBYkAAK7mi07sQAwq3Q+gQAvslgXPiD-UA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
export enum SortDirection {
 ASC = 'ASC',
 DESC = 'DESC',
}
```

### ESLint Config

```json
{
  "extends": [
    "eslint:recommended"
  ],
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/naming-convention": [ "error",
    { 
      "selector": "enumMember",
      "format":"PascalCase"
    }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

this should fail the rule, the values are all upper case, currently it catch's `FOO_BAR` and `FooBar` is fine but `FOO` is also fine.

### Actual Result

no errors

### Versions

```
❯ yarn why @typescript-eslint/eslint-plugin
└─ @cof/e1-lib@workspace:.
   └─ @typescript-eslint/eslint-plugin@npm:5.33.0 [d97b7] (via npm:^5.33.0 [d97b7])
❯ yarn why @typescript-eslint/parser
└─ @cof/e1-lib@workspace:.
   └─ @typescript-eslint/parser@npm:5.33.0 [d97b7] (via npm:^5.33.0 [d97b7])
❯ yarn why typescript
├─ @cof/e1-devexchange-client@workspace:packages/devexchange-client
│  └─ typescript@patch:typescript@npm%3A4.7.4#~builtin<compat/typescript>::version=4.7.4&hash=f456af (via patch:typescript@^4.7.4#~builtin<compat/typescript>)

Node v16.14.2
Darwin 5c52309d33e3 20.6.0 Darwin Kernel Version 20.6.0: Tue Jun 21 20:50:28 PDT 2022; root:xnu-7195.141.32~1/RELEASE_X86_64 x86_64 i386 Darwin
Yarn 3.2.2
```
please consider providing a command someone can run simply for this info ;)

**Comment by bradzacher** [comment]
This is working as expected.
If you want to enforce consecutive characters are not uppercase then use [Strict PascalCase](https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=MYewdgzgLgBAagQwDYEsAmA5BBbApjAXhgEYBuAKFElgEkM4BBAGRoBEB9AVQ1YFEAlAMoBhAPL9ehGACYKVaDDqMWrYQwAKgqQGZSQA&eslintrc=N4KABGBECmAeAu0B2ATAzpAXGA2uCU0aANgJZLyYBO0AxgPYC2jyK0Kk+AugDT6QAHYgFcA5uQzY8BKAAF4ATwFFaVUgPgBaImQqcIvflWHEiWMKBmR5SlWo3aS5eAHokAQ0blRmhkgBuyPCk9Ejm0jKEVFT0VJB8kRZg+ImQaNCmtPCx5pBsAGbuJvDxKZGQ+bGM7iVSkADK8GpZAAruaLTuxADC7dCQXGUQAL5lgyMgo0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)

https://typescript-eslint.io/rules/naming-convention#format
